.github/workflows/pr.yml

```diff
@@ -33,22 +33,30 @@ jobs:
           git checkout -b branch
           git fetch --unshallow origin main
           cmd="ansible-lint --parseable --nocolor --profile=shared"
+          unset GITHUB_ACTIONS
           $cmd > branch.output || echo "Branch linting output written"
           git checkout main
           $cmd > main.output || echo "Main linting output written"
+          export GITHUB_ACTIONS=true
           set +ex
-          diff -u0 branch.output main.output > diff.output || echo "Wrote diff file"
+          # export diff sans headers
+          diff -u0 branch.output main.output | tail -n +3 > diff.output
           echo "## Improvements over main branch:" | tee -a ${GITHUB_STEP_SUMMARY}
           echo '```diff' >> ${GITHUB_STEP_SUMMARY}
-          grep '^+[^+]' diff.output | sed -e 's/^+/+FIXED: /' | tee -a ${GITHUB_STEP_SUMMARY}
+          grep '^+' diff.output | sed -e 's/^+/+FIXED: /' | tee -a ${GITHUB_STEP_SUMMARY}
           echo '```' >> ${GITHUB_STEP_SUMMARY}
           echo "## Regressions from main branch:" | tee -a ${GITHUB_STEP_SUMMARY}
           echo '```diff' >> ${GITHUB_STEP_SUMMARY}
-          grep '^-[^-]' diff.output | sed -e 's/^-/-ERROR: /' | tee -a ${GITHUB_STEP_SUMMARY}
+          grep '^-' diff.output | sed -e 's/^-/-ERROR: /' | tee -a ${GITHUB_STEP_SUMMARY}
           echo '```' >> ${GITHUB_STEP_SUMMARY}
-          if grep -q '^-[^-]' diff.output; then
+          if grep -q '^-' diff.output; then
              echo "> Fix regressions listed above" | tee -a ${GITHUB_STEP_SUMMARY}
-             exit 1
+             regressions="$(grep '^-' diff.output | wc -l | cut -f 1 -d ' ')"
+             improvements="$(grep '^+' diff.output | wc -l | cut -f 1 -d ' ')"
+             # if ratio of regressions > improvements then fail this
```

Comment:
counting doesn't implement a quality bar as you can change the regressions without noticing.

Reply:
without counting or having a ratio this is a boolean check, either it passes 100% or it doesn't. Are you OK with this outcome?

Reply:
It should not fail more what was failing in the main branch.